internal/pkg/mqtt/client.go

```diff
@@ -342,3 +342,46 @@ func createClientOptions(
 
 	return clientOptions, nil
 }
+
+func (mc *Client) PublishBinaryData(data []byte, topic string) error {
+	optionsReader := mc.mqttClient.OptionsReader()
+	return getTokenError(
+		mc.mqttClient.Publish(
+			topic,
+			optionsReader.WillQos(),
+			optionsReader.WillRetained(),
+			data),
+		optionsReader.ConnectTimeout(),
+		PublishOperation,
+		"Unable to publish message")
+}
+
+func (mc *Client) SubscribeBinaryData(topics []types.TopicChannel, messageErrors chan error) error {
+	optionsReader := mc.mqttClient.OptionsReader()
+	for _, topic := range topics {
+		handler := newBinaryDataMessageHandler(topic.Messages)
```

Comment:
this is the only different line between the two fucntion, so please refactor to add a function `(mc *Client) subscribe(topics []types.TopicChannel, messageErrors chan error, handler pahoMqtt.MessageHandler)` to eliminate the duplicated code

Reply:
Refactored.